components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.ts

```diff
@@ -0,0 +1,221 @@
+          this.updateServerForm.controls['fqdn'].setValue(this.server.fqdn);
+          this.updateServerForm.controls['ip_address'].setValue(this.server.ip_address);
+          this.creatingServerOrg = false;
+          this.closeCreateModal();
+        });
+
+        this.store.select(deleteOrgStatus).pipe(
+          filter(status => this.id !== undefined && status === EntityStatus.loadingSuccess),
+          takeUntil(this.isDestroyed))
+          .subscribe(() => {
+            this.store.dispatch(new GetServer({ id: this.id }));
+          });
+  }
+
+  ngOnDestroy(): void {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  onSelectedTab(event: { target: { value: ChefServerTabName } }) {
+    this.tabValue = event.target.value;
+    this.router.navigate([this.url.split('#')[0]], { fragment: event.target.value });
+  }
+
+  public openCreateModal(type: string): void {
+    this.createModalVisible = true;
+    this.modalType = type;
+  }
+
+  public closeCreateModal(): void {
+    this.createModalVisible = false;
+    this.resetCreateModal();
+  }
+
+  public createServerOrg(): void {
+    this.creatingServerOrg = true;
+    const serverOrg = {
+      server_id: this.id,
+      name: this.orgForm.controls['name'].value.trim(),
+      admin_user: this.orgForm.controls['admin_user'].value.trim(),
+      admin_key: this.orgForm.controls['admin_key'].value.trim()
+    };
+    this.store.dispatch(new CreateOrg( serverOrg ));
+  }
+
+  private resetCreateModal(): void {
+    this.creatingServerOrg = false;
+    this.orgForm.reset();
+    this.conflictErrorEvent.emit(false);
+  }
+
+  public startOrgDelete(org: Org): void {
+    this.orgToDelete = org;
+    this.deleteModalVisible = true;
+  }
+
+  public deleteOrg(): void {
+    this.closeDeleteModal();
+    this.store.dispatch(new DeleteOrg(this.orgToDelete));
+  }
+
+  public closeDeleteModal(): void {
+    this.deleteModalVisible = false;
+  }
+
+  saveServer(): void {
+    this.saveSuccessful = false;
+    this.saving = true;
+    const updatedServer = {
+      id: this.server.id,
+      name: this.updateServerForm.controls.name.value.trim(),
+      description: this.updateServerForm.controls.description.value.trim(),
+      fqdn: this.updateServerForm.controls.fqdn.value.trim(),
+      ip_address: this.updateServerForm.controls.ip_address.value.trim()
+    };
+    this.store.dispatch(new UpdateServer({server: updatedServer}));
+    const pendingSave = new Subject<boolean>();
+    this.store.select(updateStatus).pipe(
+      filter(identity),
+      takeUntil(pendingSave))
+      .subscribe((state) => {
+        if (!loading(state)) {
+          pendingSave.next(true);
+          pendingSave.complete();
+          this.saving = false;
+          this.saveSuccessful = (state === EntityStatus.loadingSuccess);
+          if (this.saveSuccessful) {
+            this.updateServerForm.markAsPristine();
+          }
+        }
```

Comment:
This is a recently identified anti-pattern--we want to get rid of the `pendingSave` and the nested `subscribe`. There are many examples in the codebase introduced in the last month or two. Let me know if you need assistance in finding them.

Reply:
I applied changes as you suggested :)